docs/source/cite.rst

```diff
@@ -0,0 +1,22 @@
+Citations
```

Comment:
Perhaps `Citing Dask`? I read `Citations` as the references we cite inside our code. Note that whatever title is chosen here will also be what is in the sidebar on the doc site, so it might be nice to be more specific.